posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,572 @@
+
+# Math and ML
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+tf.get_logger().setLevel('ERROR')
+from tensorflow.keras import layers, losses, models, optimizers
+from sklearn.decomposition import PCA
+from scipy.interpolate import interp1d
+
+
+class CompileData:
+
+    def __init__(self, train_path, valid_path,
+                 profile_names=['radius','logRho', 'x_mass_fraction_H',
+                                'y_mass_fraction_He','z_mass_fraction_metals',
+                                'omega','energy']):
+        """Extracts profile data from '.h5' grid files and saves to file.
+        Args:
+            train_path (str) : path/name of '.h5' file for training data.
+            valid_path (str) : path/name of '.h5' file for testing data.
+            profile_names (array-like) : list of profile quantities to extract.
+        """
+        self.names = profile_names
+        
+        # extract testing data
+        print("extracting testing data")
+        valid = PSyGrid(valid_path)  # load PSyGrid object for testing grid
+        self.valid_scalars = pd.DataFrame()
+        self.valid_profiles = []
+        testing_failed = []
+        
+        try:
+            print(valid.final_values["S1_state"][0])
+        except:
+            print("grid does not have S1_state information")
+
+
+        for i in range(len(valid)):
+            try:
+                scalars,profiles = self.scrape(valid,i)
+                self.valid_scalars = self.valid_scalars.append(scalars,ignore_index=True)
+                self.valid_profiles.append(profiles)
+            except:
+                testing_failed.append(i)
+                pass
+        warnings.warn(f"{len(testing_failed)} binaries failed")
+
+        # extract training data
+        print("extracting training data")
+        train = PSyGrid(train_path)  # load PSyGrid object for training grid
+        self.scalars = pd.DataFrame()
+        self.profiles = []
+        training_failed = []
+        
+        for i in range(len(train)):
+            try:
+                scalars,profiles = self.scrape(train,i)
+                self.scalars = self.scalars.append(scalars,ignore_index=True)
+                self.profiles.append(profiles)
+            except:
+                training_failed.append(i)
+                pass
+        warnings.warn(f"{len(training_failed)} training binaries failed")
+        
+
+    def scrape(self,grid,ind):
+        """Extracts profile data from one MESA run.
+        Args:
+            grid (obj) : PSyGrid object.
+            ind (int) : index of run to be scraped.
+        Returns:
+            scalars (array-like) : dictionary containing initial 
+                                   m1, m2, p and final s1_state, final_m1.
+            profiles (array-like) : all N specified profiles, shape (N,200).
+        """
+        # open individual run as a DataFrame
+        df = pd.DataFrame(grid[ind]['final_profile1'])
```

Comment:
So I like the idea of using pandas df, but in IF mostly numpy structured arrays are used. I don't think this matters, but it may be worth discussing in the future, and deciding on one.

Reply:
I'll make a note to bring this and tf/torch up in a dev meeting